hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,136 @@
+defmodule Hangman.Game do
+
+  defmodule State do
+
+    #Struct for the state of the game
+    defstruct(
+      word:         "" ,
+      game_state:   :initializing,
+      turns_left:   7,
+      letters:      [],
+      used:         [],
+      last_guess:   "",
+      left_letters: MapSet.new()
+    )
+
+  end
+
+  def new_game(),             do: init_game()
+  #Only for testing
+  def new_game(word),         do: word |> init_game()
+  def tally(game),            do: game |> get_tally()
+  def make_move(game, guess), do: make_move_handler(game, guess)
+
+  ###########################
+  # End of public Functions #
+  ###########################
+
+  # Time for Private Functions
+
+  ##################################
+  # Private Functions for new_game #
+  ##################################
+  
+  # Initilize the state of the game
+  defp init_game(word \\ Dictionary.random_word()) do
+    %State{ word:         word,
+            left_letters: word |> get_letters_set(),
+            letters:      replace_letters(word) }
+  end
+
+  # Converts the random word into the MapSet
+  defp get_letters_set(word) do
+    word
+    |> String.codepoints
+    |> MapSet.new()
+  end
+
+  ###############################
+  # Private Functions for tally #
+  ###############################
+
+  # Return a map based on the game state and current status of the game
+  def get_tally(game) do
+    Map.delete(game, :word)
```

Comment:
## -2

Simply returning the internal state, minus the word, is a little dangerous. It works for now, but in the real world someone coming along later could add a field to the state and not realize that it would by default be copied out through the API.